src/definitions.rs

```diff
@@ -0,0 +1,178 @@
+use std::collections::{BTreeMap, HashMap};
+
+use nickel_lang_core::{
+    identifier::Ident,
+    term::{make, record::RecordData, LetAttrs, RichTerm, Term, UnaryOp},
+};
+use schemars::schema::Schema;
+
+use crate::{
+    contracts::{contract_from_predicate, schema_to_contract},
+    predicates::schema_to_predicate,
+};
+
+#[derive(Clone)]
+pub struct Terms {
+    predicate: RichTerm,
+    contract: RichTerm,
+}
+
+#[derive(Clone)]
+pub struct Access {
+    pub predicate: RichTerm,
+    pub contract: RichTerm,
+}
+
+#[derive(Clone)]
+pub struct EnvironmentF<T>(HashMap<String, T>);
```

Comment:
Since I introduced this, it turns out that I changed the surrounding code enough that the idea of a common environment type with different targets doesn't really make sense to have anymore. It's enough, and in fact nicer in my opinion, to make `References` the newtype around `HashMap`.